pandapower/control/util/auxiliary.py

```diff
@@ -210,41 +212,182 @@ def plot_characteristic(characteristic, start, stop, num=20, xlabel=None, ylabel
         plt.ylabel(ylabel)
 
 
-def create_trafo_characteristics(net, trafotable, trafo_index, variable, x_points, y_points):
-    # create characteristics for the specified variable and set their indices in the trafo table
-    col = f"{variable}_characteristic"
-    # check if the variable is a valid attribute of the trafo table
-    supported_columns = {"trafo": ["vk_percent_characteristic", "vkr_percent_characteristic"],
-                         "trafo3w": [f"vk{r}_{side}_percent_characteristic" for side in ["hv", "mv", "lv"] for r in
-                                     ["", "r"]]}
-    if col not in supported_columns[trafotable]:
-        raise UserWarning("Variable %s is not supported for table %s" % (variable, trafotable))
-
-    # check inputs, check if 1 trafo or multiple, verify shape of x_points and y_points and trafo_index
-    if hasattr(trafo_index, '__iter__'):
-        single_mode = False
-        if not (len(trafo_index) == len(x_points) == len(y_points)):
-            raise UserWarning("The lengths of the trafo index and points do not match!")
+def create_trafo_characteristic_object(net):
+    # check if trafo_characteristic_spline table already exists
+    if "trafo_characteristic_spline" in net:
```

Comment:
what if someone create a new trafo?

Reply:
changed the code such that, if present, the table is deleted and then re-created for all trafos from scratch.